frontend/catalyst/jax_tracer.py

```diff
@@ -1328,6 +1328,31 @@ def trace_quantum_function(
         out_tree: PyTree shapen of the result
     """
 
+    if qml.transforms.set_shots in qnode.transform_program:
```

Comment:
One thing I don't fully understand is what difference this approach makes, since the information is still static if it is in the qnode program. That is, the user is not able to set different shot values when calling their function, right?

I guess with this the user can set shots on a per qnode level, rather than a per device level?

Reply:
This is exactly what I thought as well! I don't expect this direct mutation to `device` to be the final solution at all. Just curious about any potential risk if this device mutation happens

Reply:
No risk I think _because_ each qnode triggers the creation of a new `QJITDevice`, the scoping there matches.